stackslib/src/chainstate/stacks/db/transactions.rs

```diff
@@ -9439,6 +9448,28 @@ pub mod test {
         let mut chainstate =
             instantiate_chainstate_with_balances(false, 0x80000000, function_name!(), balances);
 
+        let mut tx_runtime_checkerror_trait_no_version = StacksTransaction::new(
```

Comment:
I'm glad you added these new tests!  With the requested change to `process_transaction()` to remove the call to `StacksBlock::validate_transactions_static_epoch()`, you can preserve them simply by explicitly calling `StacksBlock::validate_transactions_static_epoch()` on these faulty transactions.